src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs

```diff
@@ -902,6 +902,36 @@ private void ProcessDependencies(PackageBuilder builder)
             }
         }
 
+        private bool FindDependency(IPackage projectPackage, IEnumerable<Tuple<IPackage, NuGet.PackageDependency>> packagesAndDependencies)
+        {
+            // returns true if the dependency should be added to the package
+            // This happens if the dependency is not a dependency of a dependecy
+            // Or if the project dependency version is > the dependency's dependency version
```

Comment:
That's a tricky one, since feeds can change we should probably assume that the minimum version does exist somewhere and not try to fit the dependencies to the current state of the feeds.